### ISSUE TITLE
Entered Password at First Launch Prompt Not Being Stored Correctly

### ISSUE THREAD

**Issue by apizz** [talkingmoose/Outlook-Exchange-Setup-5#13]
Everything appears to be working given our setup, scripts, and environment.  First launch it inputs the DOMAIN\username correctly in the prompt, however entering the user's password does not seem to get input correctly.  In Outlook preferences under accounts, all information is shown to be entered correctly but the password field is blank.  

Curious is when Outlook is then quit and reopened and the user is prompted again to enter their password _then_ the password gets stored and email immediately begins to sync.

Running on OS X 10.11.4

**Comment by strangeluck7** [comment]
We came across something similar in our environment.  I found that the script was getting into a race condition and Outlook 2016 didn't know to kick off the sync after all the account settings were set.

I was able to fix this in our environment by putting Outlook in offline mode, setting all the account properties, waiting a few seconds, then flipping it back into online mode.  I put in a pull request to add this to the script but it hasn't been looked at yet.

Essentially, this is what I added.  In the first section after:
 tell application "Microsoft Outlook"
    activate

I added this line:
set working offline to "true"

Then right after that Tell block is completed and before the "End account setup" comment:

I put this:
tell application "Microsoft Outlook"
    activate
    set working offline to "false"
end tell

That ended up making the setup script a lot more consistent in our environment.

**Comment by apizz** [comment]
Unfortunately your fix didn't work for me.  Still has the same problem.

**Comment by talkingmoose** [comment]
I've added @strangeluck7's changes (modified a little) because others have reported this has helped them.

Unfortunately, the script does not control the password prompt. When Outlook displays the prompt, the script has already finished. That doesn't mean we can't try to affect Outlook's behavior, but it's harder to do when the script doesn't trigger the password prompt in the first place.

**Comment by apizz** [comment]
@talkingmoose I tried adding a quick activate / quit tell before the Outlook tell in the AppleScript, but unfortunately that didn't work for me either.  Still have to enter password, quit, reopen, and enter password again.  Sad mac is sad :(

So to confirm, you (or anyone else) is not having this problem in your environment?  It configures the account and immediately starts syncing email after entering the password once?

Just want to figure out what's different on my end.

**Comment by talkingmoose** [comment]
I personally haven't been able to reproduce this problem. However, I'm testing against an Office 365 account on a home cable network, which may be faster to respond when Outlook attempts to connect. @strangeluck7 mentioned adding a few second delay between the end of the account setup and going online.

You might try adding "delay 5" just before the line "set working offline to false".

Let me know if that changes anything.

**Comment by csteelatgburg** [comment]
I'm seeing the same behavior as @apizz  along with #10 (creating a second Exchange account).

**Comment by talkingmoose** [comment]
Can you send me a log file privately for a user account where this is happening? Specifically, I'd like the log file to contain data from two or three logins for the same user. Thanks!

**Comment by csteelatgburg** [comment]
After working with @talkingmoose I found that I had two different versions of the LaunchAgent in /Library/LaunchAgents and they were getting in each others way. I made sure only the correct version was there and everything works great now.

**Comment by apizz** [comment]
AW YEAH.  Thank you, @talkingmoose .  Just adding that delay 5 before the "set working offline to false" did the trick.  thank you thank you.  You make the sad Mac not sad!

**Comment by talkingmoose** [comment]
After some testing, I've found a couple of unusual scenarios where two launch agents appear in the main /Library/LaunchAgents folder instead of one in the /Library/LaunchAgents folder and the other in the user's ~/Library/LaunchAgents folder. (Enable and log in to OS X's root account and install the package or install the package via command line after using sudo -s.)

Therefore, I modified the postinstall script to not run the user's launch agent as root if installing by either of those methods or something similar.

This should take care of some issues that may be causing multiple untitled accounts and/or launching Outlook automatically.

Also, I've included the five second delay before setting Outlook to work online after account creation. This may help alleviate the issue where the user is prompted for a password but Outlook doesn't accept it, requiring the user to restart Outlook.

You'll find v5.4.0 here:
https://github.com/talkingmoose/Outlook-Exchange-Setup-5

And the change log here:
https://github.com/talkingmoose/Outlook-Exchange-Setup-5/wiki/Change-log
